app/uk/gov/hmrc/gform/formtemplate/JsonSchemeValidator.scala

```diff
@@ -78,13 +77,21 @@ object JsonSchemeValidator {
                   }
                 }
 
-              val deduplicatedConditionalValidationErrors = ListSet(allConditionalValidationErrors).flatten
+              val deduplicatedConditionalValidationErrors = allConditionalValidationErrors.distinct
 
               val formattedConditionalValidationErrors: List[String] =
                 deduplicatedConditionalValidationErrors.map { case (errorProperty, errorLocation) =>
-                  s"$errorLocation: Property $errorProperty can only be used with ${conditionalRequirements(errorProperty)
-                    .mkString(", ")}"
-                }.toList
+                  val deduplicatedConditionalRequirements: List[String] = {
+                    val distinctProperties = conditionalRequirements(errorProperty).map(_.getRequiredProperty).distinct
+                    val groupedProperties = conditionalRequirements(errorProperty).groupBy(_.getRequiredProperty)
+                    distinctProperties.map(propertyKey => propertyKey -> groupedProperties(propertyKey)).map {
```

Comment:
`val distinctProperties` is used to preserve the order of required properties found in `private val conditionalRequirements`, since `groupBy` does not preserve order of keys - makes the determinism easier to predict. Happy to remove this if necessary and just use `groupBy`.